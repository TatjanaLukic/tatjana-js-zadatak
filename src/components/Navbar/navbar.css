.navbar-items {
  display: flex;
  padding: 0;
  list-style: none;
  margin: 0 0.5rem;
}
@media screen and (min-width: 576px) {
  .navbar-items {
    margin: 0;
  }
}

.navbar-item {
  box-sizing: border-box;
  display: inline-block;
  padding: 0.5rem;
  font-weight: bold;
  text-decoration: none;
  color: #ee10ee;
}
@media screen and (min-width: 576px) {
  .navbar-item {
    padding: 2rem;
    font-size: 1rem;
    text-align: center;
    border-right: 2px solid #8e0a8e;
  }
}
@media screen and (min-width: 992px) {
  .navbar-item {
    padding: 1.3rem;
    font-size: 1.3rem;
    width: 200px;
    letter-spacing: 0.1rem;
  }
}

.navbar-item--checked {
  color: white;
  background-color: #8e0a8e;
}

/*# sourceMappingURL=navbar.css.map */
