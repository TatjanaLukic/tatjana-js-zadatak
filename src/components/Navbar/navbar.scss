@import "../../assets/variables";

.navbar-items {
  display: flex;
  padding: 0;
  list-style: none;
  margin: 0 0.5rem;
  font-family: 'Roboto', sans-serif;
  //font-family:'Courier New', Courier, monospace;

  @media screen and (min-width: $smallSize) {
    margin: 0;
  }
}

.navbar-item {
  box-sizing: border-box;
  display: inline-block;
  padding: 0.5rem;
  font-weight: 700;
  text-decoration: none;
  color: $nav-main-color;
   //font-family: 'Roboto', sans-serif;
   

  @media screen and (min-width: $smallSize) {
    padding: 2rem;
    font-size: 1rem;
    text-align: center;
    border-right: 2px solid $nav-second-color;
  }

  @media screen and (min-width: $largeSize) {
    padding: 1.3rem;
    font-size: 1.3rem;
    width: 200px;
    letter-spacing: 0.1rem;
  }
}

.navbar-item--checked {
  color: $nav-active-color;
  background-color: $nav-second-color;
}
