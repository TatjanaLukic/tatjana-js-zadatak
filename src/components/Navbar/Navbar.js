import React from "react";
import { NavLink } from "react-router-dom";

import "./navbar.css";

const Navbar = () => {
  return (
    <ul className="navbar-items">
      <li>
        <NavLink
          to="/TopNewsPage"
          className="navbar-item"
          activeClassName="navbar-item--checked"
        >
          Top News
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/CategoriesPage"
          className="navbar-item"
          activeClassName="navbar-item--checked"
        >
          Categories
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/SearchPage"
          className="navbar-item"
          activeClassName="navbar-item--checked"
        >
          Search
        </NavLink>
      </li>
    </ul>
  );
};

Navbar.propTypes = {};

export default Navbar;
