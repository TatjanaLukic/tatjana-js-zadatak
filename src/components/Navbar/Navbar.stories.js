import React from "react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "./Navbar";

export default {
  title: "App/Navbar",
  component: Navbar,
  decorators: [(getStory) => <MemoryRouter>{getStory()}</MemoryRouter>],
};

const Template = () => <Navbar />;

export const Default = Template.bind({});
