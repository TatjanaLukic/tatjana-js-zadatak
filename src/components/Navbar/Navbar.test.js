import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Navbar from "../Navbar/Navbar";
import { MemoryRouter } from "react-router-dom";

describe("<Navbar />", () => {
  it("should render three NavLinks", () => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    const links = screen.getAllByRole("link");
    expect(links).toHaveLength(3);
   
    expect(links[0]).toHaveAttribute("href", "/TopNewsPage");
    expect(links[0].textContent).toBe("Top News");

    expect(links[1]).toHaveAttribute("href", "/CategoriesPage");
    expect(links[1].textContent).toBe("Categories");

    expect(links[2]).toHaveAttribute("href", "/SearchPage");
    expect(links[2].textContent).toBe("Search");

    expect(links[0].className).toBe("navbar-item");
    fireEvent.click(links[0]);
    expect(links[0].className).toBe("navbar-item navbar-item--checked");

  });

  it("spanshots matche", () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    const links = screen.getAllByRole("link");
    expect(container).toMatchSnapshot();

    fireEvent.click(links[0]);
    expect(container).toMatchSnapshot();
  });
});
