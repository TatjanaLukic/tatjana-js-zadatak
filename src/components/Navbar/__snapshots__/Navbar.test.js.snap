// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> spanshots matche 1`] = `
<div>
  <ul
    class="navbar-items"
    data-testid="navbar"
  >
    <li>
      <a
        class="navbar-item"
        href="/TopNewsPage"
      >
        Top News
      </a>
    </li>
    <li>
      <a
        class="navbar-item"
        href="/CategoriesPage"
      >
        Categories
      </a>
    </li>
    <li>
      <a
        class="navbar-item"
        href="/SearchPage"
      >
        Search
      </a>
    </li>
  </ul>
</div>
`;

exports[`<Navbar /> spanshots matche 2`] = `
<div>
  <ul
    class="navbar-items"
    data-testid="navbar"
  >
    <li>
      <a
        aria-current="page"
        class="navbar-item navbar-item--checked"
        href="/TopNewsPage"
      >
        Top News
      </a>
    </li>
    <li>
      <a
        class="navbar-item"
        href="/CategoriesPage"
      >
        Categories
      </a>
    </li>
    <li>
      <a
        class="navbar-item"
        href="/SearchPage"
      >
        Search
      </a>
    </li>
  </ul>
</div>
`;
